### ISSUE TITLE
2.2.105.toolpath.sentinel directory missing ?

### ISSUE THREAD

**Issue by dagood** [dotnet/cli#11049]
@DweeberlyLoom commented on [Wed Mar 20 2019](https://github.com/dotnet/core/issues/2483)

Problem encountered on https://dotnet.microsoft.com/learn/web/aspnet-microservice-tutorial/create
Operating System: windows

```
PS C:\Users\dweeb> dotnet new webapi -o myMicroservice --no-https
System.UnauthorizedAccessException: Access to the path 'C:\Users\dweeb\.dotnet\2.2.105.toolpath.sentinel' is denied.
   at System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle)
   at System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   at System.IO.File.Create(String path)
   at Microsoft.Extensions.EnvironmentAbstractions.FileWrapper.CreateEmptyFile(String path)
   at Microsoft.DotNet.Configurer.FileSentinel.Create()
   at Microsoft.DotNet.Configurer.DotnetFirstTimeUseConfigurer.Configure()
   at Microsoft.DotNet.Cli.Program.ConfigureDotNetForFirstTimeUse(INuGetCacheSentinel nugetCacheSentinel, IFirstTimeUseNoticeSentinel firstTimeUseNoticeSentinel, IAspNetCertificateSentinel aspNetCertificateSentinel, IFileSentinel toolPathSentinel, Boolean hasSuperUserAccess, DotnetFirstRunConfiguration dotnetFirstRunConfiguration, IEnvironmentProvider environmentProvider)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)
PS C:\Users\dweeb> dir 'C:\Users\dweeb\.dotnet\2.2.105*'


    Directory: C:\Users\dweeb\.dotnet

Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        3/20/2019   9:01 PM              5 2.2.105_IsDockerContainer.dotnetUserLevelCache
-a----        3/20/2019   9:01 PM             64 2.2.105_MachineId.dotnetUserLevelCache
```

It would appear that the SDK doesn't install what you think it installs ... ?

---

@karelz commented on [Sat Mar 30 2019](https://github.com/dotnet/core/issues/2483#issuecomment-478274906)

@DweeberlyLoom how did you install the SDK?
Is it maybe some weird machine setup? Can you try it on clean VM/another machine to see if it is one machine problem or general problem? Thanks!

---

@karelz commented on [Sat Mar 30 2019](https://github.com/dotnet/core/issues/2483#issuecomment-478274928)

cc @dagood 

---

@dagood commented on [Mon Apr 01 2019](https://github.com/dotnet/core/issues/2483#issuecomment-478623911)

Moving to the dotnet/cli repo because `Microsoft.DotNet.Configurer.FileSentinel` lives there, and I think the people with the most context will see it there.

So far, this does look like an unusual permissions issue: `System.UnauthorizedAccessException: Access to the path 'C:\Users\dweeb\.dotnet\2.2.105.toolpath.sentinel' is denied.`

**Comment by DweeberlyLoom** [comment]
@livarcocc 
Thank you for your reply.  I installed the SDK as per the instructions found here: https://dotnet.microsoft.com/learn/web/aspnet-microservice-tutorial/install  I used the 64 bit SDK.

I had not previously noticed that the path was being created.  As you have already assessed the \users\dweeb\.dotnet directory did not allow the builtin users group to create files/dirs.  It's filesystem rights were set to ReadAndExecute, Synchronize.  The .dotnet directory was not something I created and I don't know how/why it was created with those ACLs.

I corrected this by giving full control rights to the .dotnet directory.  After doing so the given commands worked as expected.
